crates/forge_api/src/executor.rs

```diff
@@ -52,4 +50,22 @@ impl<F: Infrastructure + App> ForgeExecutorService<F> {
             }
         }))
     }
+
+    pub async fn retry(
+        &self,
+        conversation_id: ConversationId,
+    ) ->anyhow::Result<MpscStream<anyhow::Result<AgentMessage<ChatResponse>>>>{
+        let conversation = self
+        .app
+        .conversation_service()
+        .get(&conversation_id)
+        .await?
+        .ok_or(Error::ConversationNotFound(conversation_id.clone()))?;
+    let last_user_message = conversation
+        .rfind_event(Event::USER_TASK_UPDATE)
+        .or_else(|| conversation.rfind_event(Event::USER_TASK_INIT))
+        .ok_or(anyhow::anyhow!("No user message found in the conversation"))?;
```

Comment:
Thinking about this again. I think we should just simply retry, and not from the last user message. Sometimes the agent can be in the middle of tool calls etc and there could be some failures so we should retry from that step instead of the first user message.

Reply:
I saw the issues there was one issue related to that, should we target that issue in this PR as well?